Android/app/src/main/java/uncommon/common/activity/ReservationActivity.java

```diff
@@ -109,7 +129,28 @@ protected void onCreate(Bundle savedInstanceState) {
             timeSlotIdxList.add(timeslot.get(timeListIdx).getTimeTableIdx());
         }
 
-        timeListView.setAdapter(adapter);
+        // grey out on time slot
+        final ArrayAdapter<String> timeslotAdapter = new ArrayAdapter<String>
+                (this, android.R.layout.simple_list_item_1, timeList){
+            @Override
+            public View getView(int position, View convertView, ViewGroup parent){
+                // Get the current item from ListView
+                View view = super.getView(position, convertView, parent);

```

Comment:
I think (I may be wrong) you can cast view to TextView, so you don't need to the casting below.